pkg/kubectl/kubectl.go

```diff
@@ -46,6 +46,31 @@ func makeImageList(spec *api.PodSpec) string {
 	return strings.Join(listOfImages(spec), ",")
 }
 
+func NewThirdPartyResourceMapper(gvs []unversioned.GroupVersion, gvks []unversioned.GroupVersionKind) (meta.RESTMapper, error) {
+	mapper := meta.NewDefaultRESTMapper(gvs, func(gv unversioned.GroupVersion) (*meta.VersionInterfaces, error) {
```

Comment:
Actually I think you should use the same function. The one @deads2k pasted looks correct. The one you wrote here seems to have a wrong `ObjectConvertor`.